pdns/zoneparser-tng.cc

```diff
@@ -107,14 +107,19 @@ static string makeString(const string& line, const pair<string::size_type, strin
 
 static bool isTimeSpec(const string& nextpart)
 {
-  if(nextpart.empty())
+  if (nextpart.empty()) {
     return false;
-  for(string::const_iterator iter = nextpart.begin(); iter != nextpart.end(); ++iter) {
-    if(isdigit(*iter))
+  }
+  for (auto iter = nextpart.begin(); iter != nextpart.end(); ++iter) {
+    if (isdigit(*iter) != 0) {
```

Comment:
to be safe, `*iter` should be cast to `unsiged char`. See https://en.cppreference.com/w/cpp/string/byte/tolower

Reply:
Thanks will finish processing review later this morning it's really interresting thanks for your time. I have somes holes with Cpp rust and C are OK.

Will take some time to check what exactly clang-tidy suggested hope it's my error and not the suggestion